serverless/functions/join-stream-as-speaker.js

```diff
@@ -57,9 +56,26 @@ module.exports.handler = async (context, event, callback) => {
     return callback(null, response);
   }
 
+  // Fetch stream map item
+  try {
+    const backendStorageSyncClient = await client.sync.services(context.BACKEND_STORAGE_SYNC_SERVICE_SID);
+    streamMapItem = await backendStorageSyncClient.syncMaps('streams').syncMapItems(room.sid).fetch();
+  } catch (e) {
+    response.setStatusCode(500);
+    response.setBody({
+      error: {
+        message: 'error finding stream map item',
+        explanation: e.message,
+      },
+    });
+    return callback(null, response);
+  }
```

Comment:
Might be worth adding this to the common code

https://github.com/twilio/twilio-live-interactive-video/blob/main/serverless/middleware/common.private.js

Reply:
I pushed some changes for this. I think it's a small step in the right direction. It does reduce duplicate code a little. What I don't like about it is:

1. I don't have a way to reuse the error handling which is the bulk of the code. I think the large amount of error handling boilerplate makes it difficult to follow the backend code.
2. Having all reusable code in a single file does not seem like it will scale well. I would prefer to have something like `backend-storage.private.js` for anything related to backend storage. And ideally it completely abstracts the fact that it is backed by sync.

I have ideas but I think I need to team up with a JS pro before going down this path more so probably best to wait until Tim M is back.

Reply:
Fair enough. Thanks for reducing some of the duplicate code!